pyvista/core/grid.py

```diff
@@ -918,7 +918,7 @@ def _generate_rectilinear_coords(
         # Use linspace to avoid rounding error accumulation
         dims = self.dimensions
         spacing = self.spacing
-        origin = self.origin
+        origin = np.array(self.origin) + np.array(self.offset)
         return [
             (np.linspace(0, (dims[i] - 1) * spacing[i], dims[i]) + origin[i]) for i in range(3)
         ]
```

Comment:
Offset is in IJK coordinates, so to convert to XYZ coords you must include the spacing.
```suggestion
        origin = self.origin
        offset = self.offset
        return [
            (np.linspace(offset[i] * spacing[i], (offset[i] + dims[i] - 1) * spacing[i], dims[i]) + origin[i]) for i in range(3)
        ]
```

And if we're including offset, we should also include the direction matrix. I think we can break this up into several statements:
``` suggestion
        origin = self.origin
        offset = self.offset
        direction = self.direction_matrix

        # Grid IJK indices
        ijk = [np.linspace(offset[i], offset[i] + dims[i] - 1, dims[i]) for i in range(3)]

        # Apply the transformation: coord = origin + direction @ (index * spacing)
        return [sum(direction[axis, j] * ijk[j] for j in range(3)) + origin[axis] for axis in range(3)]
```